d3ploy/solver.py

```diff
@@ -172,3 +172,47 @@ def minimize_number_of_deployment(proto_commod, remainder):
         break
 
     return deploy_dict
+
+def find_mins(commod_min, commod_dict):
+    """ This function updates the commod_min
+    dictionary to contain the minimum production
+    for each commodity. 
+     Parameters:
+    ----------
+    commod_min: dictionary
+        key:commodity
+        value: min capacity
+        dictionary of the minimum commod production    
+    commod_dict: dictionary
+        key: prototype name
+        value: prototype capacity
+    
+    Returns:
+    --------
+    commod_min: dictionary
+        key: commodity
+        value: min capacity
+    """
+    for commod, proto in commod_dict.items(): 
+        commod_min[commod] = 1e299
+        for fac, dic in proto.items():
+            if dic['cap'] < commod_min[commod]:
+                commod_min[commod] = dic['cap']
+    return commod_min
+
+def decommission(agent, commod_dict, diff, commod):
+    fac_mins = {}
+    for fac, dic in commod_min[commod]:
```

Comment:
This function will fail. `commod_min` is not defined in this scope. 
Maybe you meant `commod_dict` ...